src/Texy/modules/BlockModule.php

```diff
@@ -134,41 +130,41 @@ public function solve($invocation, $blocktype, $s, $param, $mod)
 		}
 
 		if ($blocktype === 'block/code') {
-			$s = Texy::outdent($s);
+			$s = Texy\Texy::outdent($s);
 			if ($s === '') {
 				return "\n";
 			}
-			$s = Texy::escapeHtml($s);
-			$s = $tx->protect($s, Texy::CONTENT_BLOCK);
-			$el = TexyHtml::el('pre');
+			$s = Texy\Texy::escapeHtml($s);
+			$s = $tx->protect($s, Texy\Texy::CONTENT_BLOCK);
+			$el = Texy\HtmlElement::el('pre');
 			$mod->decorate($tx, $el);
-			$el->attrs['class'][] = $param; // lang
-			$el->create('code', $s);
+			$c = $el->create('code', $s);
+			$c->attrs['class'][] = "language-$param"; // lang
```

Comment:
This moves language class to code. Why? I have issues with SyntaxHighlighter, maybe this would fix it.

Reply:
Semantically, it is language of code, not pre.

Reply:
Pardon, pisu anglicky ze zvyku. Mmch nekde jsem cetl, ze je "hard to get to the community for non-czech speakers". :)

Nevíš prosím, je někde nějaké option jak vyhodit <code> pod <pre> ? Není to kompatibilní se SyntaxHighlighter. Je jediná cesta vlastní block handler?

Reply:
Už to asi mám.

```
$texy->addHandler('block', array('removeCodeFromPre', 'blockHandler'));

class removeCodeFromPre
{
    /**
     * @param TexyHandlerInvocation  handler invocation
     * @param string  block type
     * @param string  text to highlight
     * @param string  language
     * @param TexyModifier modifier
     * @return TexyHtml
     */
    function blockHandler($invocation, $blocktype, $s, $lang, $modifier)
    {
        if ($blocktype !== 'block/code')
            return $invocation->proceed();

        $tx = $this->texy;
        $s = Texy\Texy::outdent($s);
        if ($s === '') {
            return "\n";
        }
        $s = Texy\Texy::escapeHtml($s);
        $s = $tx->protect($s, Texy\Texy::CONTENT_BLOCK);
        $el = Texy\HtmlElement::el('pre');
        $modifier->decorate($tx, $el);
        //$c = $el->create('code', $s);
        $el->attrs['class'][] = "language-$lang";
        return $el;
    }
}
```